core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/upscale/StackUpscaleActions.java

```diff
@@ -119,14 +124,36 @@ protected void prepareExecution(StackScaleTriggerEvent payload, Map<Object, Obje
 
             @Override
             protected void doExecute(StackScalingFlowContext context, StackScaleTriggerEvent payload, Map<Object, Object> variables) {
-                Map<String, Integer> hostGroupsWithAdjustment = payload.getHostGroupsWithAdjustment();
+                if (context.isRepair()) {
+                    LOGGER.debug("We do not need to update domainDnsResolver in case of repair activity, since it matters only in case of upscale activity.");
+                    sendEvent(context, new UpdateDomainDnsResolverResult(context.getStack().getId()));
+                } else {
+                    sendEvent(context, new UpdateDomainDnsResolverRequest(context.getStack().getId()));
+                }
+            }
+
+            @Override
+            protected Object getFailurePayload(StackScaleTriggerEvent payload, Optional<StackScalingFlowContext> flowContext, Exception ex) {
+                return new StackFailureEvent(EventSelectorUtil.failureSelector(UpdateDomainDnsResolverResult.class), payload.getResourceId(), ex);
+            }
+        };
+    }
+
+    @Bean(name = "UPSCALE_PREVALIDATION_STATE")
+    public Action<?, ?> prevalidate() {
+        return new AbstractStackUpscaleAction<>(UpdateDomainDnsResolverResult.class) {
+
+            @Override
+            protected void doExecute(StackScalingFlowContext context, UpdateDomainDnsResolverResult payload, Map<Object, Object> variables) {
```

Comment:
I can see payload was changed. If an old UPSCALE_PREVALIDATION_STATE is restarted then the payload will be StackScaleTriggerEvent

Reply:
looks like flow engine can handle this:

application on master stopped at validation state:
![Screenshot from 2022-03-30 12-44-23](https://user-images.githubusercontent.com/26329499/160815412-7e8b4227-9a3f-4a4e-ac51-01524f6240b4.png)
![Screenshot from 2022-03-30 12-44-59](https://user-images.githubusercontent.com/26329499/160815420-83d35c9a-c15c-4f88-917f-654501b0f344.png)

then application restarted on this branch:

![Screenshot from 2022-03-30 12-50-41](https://user-images.githubusercontent.com/26329499/160815570-950c7bcc-0c18-405a-b4a2-fcdbc18fe4a5.png)
![Screenshot from 2022-03-30 12-50-56](https://user-images.githubusercontent.com/26329499/160815578-ed9e2ea2-ed39-4fd3-8d43-2bf0a066c789.png)